### PR TITLE
New Package: py-cherrypy

### PR DESCRIPTION
**Pull request by marcmengel**
cherrypy web framework.

Depends on #13582, #13583, #13584, #13585, #13586

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-cherrypy/package.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from spack import *
+
+class PyCherrypy(PythonPackage):
+    """CherryPy is a pythonic, object-oriented HTTP framework."""
+
+    homepage = "https://cherrypy.org/"
+    url      = "https://files.pythonhosted.org/packages/1f/de/3327bd7168be762180924085fecef2e127d128f1d6157f88cd87fdac2971/CherryPy-18.1.1.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cherrypy/package.py]:
```suggestion
    url      = "https://pypi.io/packages/source/C/CherryPy/CherryPy-18.1.1.tar.gz"
```

File: var/spack/repos/builtin/packages/py-cherrypy/package.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    version('18.1.1', '76e5d3c6b7be845345f871c604cfdf58')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-docutils', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cherrypy/package.py]: This dep is only needed to build the docs, remove it.

File: var/spack/repos/builtin/packages/py-cherrypy/package.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    depends_on('py-more-itertools',    type=('build', 'run'))
+    depends_on('py-zc-buildout',       type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cherrypy/package.py]: Not a real dependency, remove it

File: var/spack/repos/builtin/packages/py-cherrypy/package.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    depends_on('py-zc-lockfile',       type=('build', 'run'))
+    depends_on('py-cheroot',           type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cherrypy/package.py]:
```suggestion
    depends_on('py-cheroot@6.2.4:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-cherrypy/package.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    depends_on('py-tempora',           type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cherrypy/package.py]: Not a dep, remove it

File: var/spack/repos/builtin/packages/py-cherrypy/package.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    depends_on('py-portend',           type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cherrypy/package.py]:
```suggestion
    depends_on('py-portend@2.1.1:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-cherrypy/package.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+class PyCherrypy(PythonPackage):
+    """CherryPy is a pythonic, object-oriented HTTP framework."""
+
+    homepage = "https://cherrypy.org/"
+    url      = "https://files.pythonhosted.org/packages/1f/de/3327bd7168be762180924085fecef2e127d128f1d6157f88cd87fdac2971/CherryPy-18.1.1.tar.gz"
+
+    version('18.1.1', '76e5d3c6b7be845345f871c604cfdf58')
+
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cherrypy/package.py]: Missing build dep on `py-setuptools-scm`.

File: var/spack/repos/builtin/packages/py-cherrypy/package.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
+from spack import *
+
+class PyCherrypy(PythonPackage):
+    """CherryPy is a pythonic, object-oriented HTTP framework."""
+
+    homepage = "https://cherrypy.org/"
+    url      = "https://files.pythonhosted.org/packages/1f/de/3327bd7168be762180924085fecef2e127d128f1d6157f88cd87fdac2971/CherryPy-18.1.1.tar.gz"
+
+    version('18.1.1', '76e5d3c6b7be845345f871c604cfdf58')
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cherrypy/package.py]: Missing dep on Python 3.5+

Comment (marcmengel) [reply var/spack/repos/builtin/packages/py-cherrypy/package.py]: Is there some way to say that 1.8 requires 3.5+, but 1.7 doesn't?

File: var/spack/repos/builtin/packages/py-cherrypy/package.py (hunk 5 above)
Comment (marcmengel) [reply var/spack/repos/builtin/packages/py-cherrypy/package.py]: Sigh, portend depends on tempora, nto cherrypy directly...

File: var/spack/repos/builtin/packages/py-cherrypy/package.py (hunk 8 above)
Comment (marcmengel) [reply var/spack/repos/builtin/packages/py-cherrypy/package.py]: Found it...

File: var/spack/repos/builtin/packages/py-cherrypy/package.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    depends_on('py-setuptools',     type='build')
+    depends_on('py-setuptools-scm', type='build')
+    depends_on('py-more-itertools', type=('build', 'run'))
+    depends_on('py-zc-buildout',    type=('build', 'run'))
+    depends_on('py-zc-lockfile',    type=('build', 'run'))
+    depends_on('py-cheroot@6.2.4:', type=('build', 'run'))
+    depends_on('py-portend@2.1.1:', type=('build', 'run'))
+    depends_on('python@3.5:',       when='@18.0.0:',  type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cherrypy/package.py]: You only have one version, so you could drop the `when` if you want to make the package simpler.

File: var/spack/repos/builtin/packages/py-cherrypy/package.py [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+    homepage = "https://cherrypy.org/"
+    url      = "https://pypi.io/packages/source/C/CherryPy/CherryPy-18.1.1.tar.gz"
+
+    version('18.1.1', '76e5d3c6b7be845345f871c604cfdf58')
+
+    depends_on('py-setuptools',     type='build')
+    depends_on('py-setuptools-scm', type='build')
+    depends_on('py-more-itertools', type=('build', 'run'))
+    depends_on('py-zc-buildout',    type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cherrypy/package.py]: Not required, remove

**Comment by marcmengel** [comment]
@adamjstewart where do you get those pretty urls for the source packages?

File: var/spack/repos/builtin/packages/py-cherrypy/package.py (hunk 9 above)
Comment (marcmengel) [reply var/spack/repos/builtin/packages/py-cherrypy/package.py]: Well that way it's clearer you can pull the 17.x version if you want python 2.7 support...

**Comment by adamjstewart** [comment]
@marcmengel see https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#url

File: var/spack/repos/builtin/packages/py-cherrypy/package.py [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+from spack import *
+
+
+class PyCherrypy(PythonPackage):
+    """CherryPy is a pythonic, object-oriented HTTP framework."""
+
+    homepage = "https://cherrypy.org/"
+    url      = "https://pypi.io/packages/source/C/CherryPy/CherryPy-18.1.1.tar.gz"
+
+    version('18.1.1', '76e5d3c6b7be845345f871c604cfdf58')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cherrypy/package.py]:
```suggestion
    version('18.1.1', sha256='6585c19b5e4faffa3613b5bf02c6a27dcc4c69a30d302aba819639a2af6fa48b')
```

**Comment by marcmengel** [comment]
Thanks for walking me through this ... sorry for the multiple rounds.

**Comment by adamjstewart** [comment]
No worries, thanks for contributing all of these packages!

**Pull request merged**